libs/providers/flagd/src/lib/service/grpc/service.ts

```diff
@@ -0,0 +1,107 @@
+import {IService} from '../IService'
+import {ServiceClient} from '../../../proto/ts/schema/v1/schema.client'
+import * as grpc from '@grpc/grpc-js';
+import {
+    EvaluationContext,
+    ResolutionDetails
+} from '@openfeature/nodejs-sdk'
+import {
+    ResolveBooleanRequest,
+    ResolveStringRequest,
+    ResolveNumberRequest,
+    ResolveObjectRequest
+} from '../../../proto/ts/schema/v1/schema'
+import { RpcError } from '@protobuf-ts/runtime-rpc'
+import { Struct } from '../../../proto/ts/google/protobuf/struct'
+import client from './client'
+
+export class GRPCService implements IService {
+    client: ServiceClient;
+    constructor(host?: string, port?: number, credentials?: grpc.ChannelCredentials) {
+        this.client = new client(host, port, credentials)
+    }
+
+
+    async ResolveBoolean(flagKey: string, defaultValue: boolean, context: EvaluationContext): Promise<ResolutionDetails<boolean>> {
+        const res: ResolutionDetails<boolean> = {
+            value: defaultValue
+        }
+        const req: ResolveBooleanRequest = {
+            flagKey,
+            context: Struct.fromJsonString(JSON.stringify(context)),
+        }
+        await this.client.resolveBoolean(req, undefined).response.then(callRes => {
```

Comment:
I would recommend not merging async/await with promises like this. With await, you can refactor this to something like this:

```typescript 
try {
    try {
      const { response } = await this.client.resolveBoolean({
        flagKey,
        context: Struct.fromJsonString(JSON.stringify(context)),
      });
      return {
        value: response.value,
        reason: response.reason,
        variant: response.variant,
      };
    } catch (err: unknown) {
      return {
        reason: 'ERROR',
        errorCode: (err as Partial<RpcError>)?.code ?? StandardResolutionReasons.UNKNOWN,
        variant: 'default_value',
        value: defaultValue,
      }
    }
}
```

Reply:
refactored and mirrored in the http implementation